include/envoy/buffer/buffer.h

```diff
@@ -32,6 +57,13 @@ class Instance {
    */
   virtual void add(const void* data, uint64_t size) PURE;
 
+  /**
+   * Add externally owned data into the buffer. No copying is done. fragment is not owned. When
+   * the fragment->data() is no longer needed, fragment->done() is called.
+   * @param fragment the externally owned data to add to the buffer.
+   */
+  virtual void addBufferFragment(BufferFragment* fragment) PURE;
```

Comment:
Nit: I would pass as `BufferFragment&` as the fragment cannot be nullptr.

Reply:
Done.